aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1095,9 +1094,16 @@ Tensor &any_out(const Tensor &self, int64_t dim, bool keepdim, Tensor &result) {
 }
 
 Tensor &amin_out(const Tensor& self, IntArrayRef dim, bool keepdim, Tensor& result) {
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:",
+              self.scalar_type(), result.scalar_type());
```

Comment:
I... guess this CHECK is alright, but I'm not really sure it's in this PR (and it doesn't seem to have been consistently applied at other sites)

Reply:
Makes sense removed it.